model/validate.go

```diff
@@ -7,9 +7,31 @@ import (
 	"net/http"
 	"strings"
 
+	"fmt"
 	"github.com/vmware/virtual-security-module/util"
+	"reflect"
 )
 
+func (s *SecretEntry) Equal(t *SecretEntry) (bool, error) {
```

Comment:
returning both bool and error is a bit redundant:
(1) if you want to return an informative error, then you don't need to return a bool, because err == nil if and only if bool == true.
(2) If you just want to know if the two structures are Equal just return a bool.

Personally I would go with the second option here.